raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -0,0 +1,204 @@
+
+        this.logEndEpochMetricName = metrics.metricName("log-end-epoch", metricGroupName, "The current raft log end epoch.");
+        metrics.addMetric(this.logEndEpochMetricName, (mConfig, currentTimeMs) -> logEndEpoch);
+
+        this.bootTimestampMetricName = metrics.metricName("boot-timestamp", metricGroupName, "The bootstrapped timestamp of this member.");
+        metrics.addMetric(this.bootTimestampMetricName, (mConfig, currentTimeMs) -> bootTimestamp);
+
+        this.numVoterConnectionsMetricName = metrics.metricName("number-voter-connections", metricGroupName, "The number of voter connections recognized at this member.");
+        metrics.addMetric(this.numVoterConnectionsMetricName, (mConfig, currentTimeMs) -> numVoterConnections);
+
+        this.pollIdleSensor = metrics.sensor("poll-idle-ratio");
+        this.pollIdleSensor.add(metrics.metricName("poll-idle-ratio-avg",
+                metricGroupName,
+                "The average fraction of time the consumer's poll() is idle as opposed to waiting for the user code to process records."),
+                new Avg());
+
+        this.commitTimeSensor = metrics.sensor("commit-latency");
+        this.commitTimeSensor.add(metrics.metricName("commit-latency-avg", metricGroupName,
+                "The average time in milliseconds to commit an entry in the raft log."), new Avg());
+        this.commitTimeSensor.add(metrics.metricName("commit-latency-max", metricGroupName,
+                "The maximum time in milliseconds to commit an entry in the raft log."), new Max());
+
+        this.electionTimeSensor = metrics.sensor("election-latency");
+        this.electionTimeSensor.add(metrics.metricName("election-latency-avg", metricGroupName,
+                "The average time in milliseconds to elect a new leader."), new Avg());
+        this.electionTimeSensor.add(metrics.metricName("election-latency-max", metricGroupName,
+                "The maximum time in milliseconds to elect a new leader."), new Max());
+
+        this.fetchRecordsSensor = metrics.sensor("fetch-records");
+        this.fetchRecordsSensor.add(metrics.metricName("fetch-records-rate", metricGroupName,
+                "The average number of records fetched from the leader of the raft quorum."),
+                new Rate(TimeUnit.SECONDS, new WindowedSum()));
+
+        this.appendRecordsSensor = metrics.sensor("append-records");
+        this.appendRecordsSensor.add(metrics.metricName("append-records-rate", metricGroupName,
+                "The average number of records appended per sec as the leader of the raft quorum."),
+                new Rate(TimeUnit.SECONDS, new WindowedSum()));
+    }
+
+    public void updatePollStart(long currentTimeMs) {
+        if (pollEndMs != 0L) {
+            long pollTimeMs = Math.max(pollEndMs - pollStartMs, 0L);
+            long totalTimeMs = Math.max(currentTimeMs - pollStartMs, 1L);
```

Comment:
We could make `1L` a static final variable indicating the measurement granularity.